tests/models/jit_test.py

```diff
@@ -0,0 +1,105 @@
+import torch
+import pytest
+from torch.testing import assert_allclose
+from asteroid.models import DeMask, ConvTasNet, DPRNNTasNet, DPTNet, LSTMTasNet
+
+
+@pytest.fixture(scope="module")
+def small_model_params():
+    params = {
+        ConvTasNet.__name__: {
+            "n_src": 2,
+            "n_repeats": 2,
+            "n_blocks": 3,
+            "bn_chan": 16,
+            "hid_chan": 4,
+            "skip_chan": 8,
+            "n_filters": 32,
+        },
+        DPRNNTasNet.__name__: {
+            "n_src": 2,
+            "n_repeats": 2,
+            "bn_chan": 16,
+            "hid_size": 4,
+            "chunk_size": 20,
+            "n_filters": 32,
+        },
+        DPTNet.__name__: {
+            "n_src": 2,
+            "ff_hid": 10,
+            "chunk_size": 4,
+            "n_repeats": 2,
+        },
+        LSTMTasNet.__name__: {
+            "n_src": 2,
+            "hid_size": 4,
+            "n_layers": 1,
+            "n_filters": 32,
+            "dropout": 0.0,
+        },
+        DeMask.__name__: {
+            "input_type": "mag",
+            "output_type": "mag",
+            "hidden_dims": [64],
+            "dropout": 0,
+            "activation": "relu",
+            "mask_act": "relu",
+            "norm_type": "gLN",
+            "n_filters": 8,
+            "stride": 4,
+            "kernel_size": 8,
+        },
+    }
+
+    return params
+
+
+def test_enhancement_model(small_model_params):
+    params = small_model_params["DeMask"]
+    filter_banks = ["free", "stft", "analytic_free", "param_sinc"]
```

Comment:
Let's test the filterbanks independently of the models, otherwise it'll grow too much. 
By using `BaseModel` with a `nn.Identity` masker, and test consistency with different input shapes once traced.

Reply:
Do you mean having a class inheriting from `BaseModel` with a `masker` component set to `torch.nn.Identity` just to check that the traced model can process input data with various values of `ndim`?

In this case, why not a class inheriting from `BaseEncoderMaskerDecoder` instead? It makes writing the class much easier.

I have written the test but, as suspected, tracing will only take one of the branches in the `forward`s of the encoder and the decoder meaning that if a model has been traced with, say, a 2D input, the traced model will only work with 2D tensors. From here, we can either script the `forward`s of the encoder and decoder or rewrite these methods to not have branches. I'll have a look at the scripting solution.

Reply:
I meant  `BaseEncoderMaskerDecoder`, yes. 
And yes, scripting will be the solution.